src/main/java/de/eposcat/master/approachImpl/EAV_DatabaseAdapter.java

```diff
@@ -6,150 +6,170 @@
+            st.setLong(2, page.getId());
             int affectedRows = st.executeUpdate();
-            
+
             if(affectedRows != 1) {
-                throw new SQLException();
+                throw new BlException("page is not tracked by database, try create pageWithAttributes first");
             }
-            
-            
-            for (String attributename : page.getAttributes().keySet()) {
-                page.getAttributes().put(attributename, saveAttribute(attributename, page.getAttributes().get(attributename)));
+
+            //TODO Dont change while iterating
+            HashMap<String, Attribute> persistedAttributesWithIds = new HashMap<>();
+            for (String attributeName : page.getAttributes().keySet()) {
+                persistedAttributesWithIds.put(attributeName, saveAttribute(attributeName, page.getAttributes().get(attributeName)));
             }
-            
-            final List<Integer> currentAttributes = getCurrentAttributeIds(page.getId());
-            
-            page.getAttributes().values().stream().forEach(att -> {
-                try {
-                    int attId = att.getId();
-                    if(currentAttributes.contains(attId)) {
-                        currentAttributes.remove(attId);
-                        updateAttributeValue(page.getId(), att);
-                    } else {
-                        createAttributeValue(page.getId(), att);
-                    }
-                } catch (SQLException e) {
-                    //Java stream interface forces us to handle the exception inside the stream
-                    //TODO Better exception handling for streams?
-                    e.printStackTrace();
-                    System.out.println("Couldn't update value of AttributeValue with id: " + att.getId());
-                }
-                
-            });
-            
-            currentAttributes.stream().forEach(id -> {
-                try {
-                    removeAttributeValue(page.getId(), id);
-                } catch (SQLException e) {
-                    e.printStackTrace();
-                    System.out.println("Couldn't remove AttributeValue with id: " + id);
+
+            page.setAttributes(persistedAttributesWithIds);
+
+            final List<Long> currentAttributes = getLastAttributeIds(page.getId());
+
+            for(Attribute att : page.getAttributes().values()){
+                long attId = att.getId();
+                if(currentAttributes.contains(attId)) {
+                    currentAttributes.remove(attId);
+                    updateAttributeValue(page.getId(), att);
+                } else {
+                    createAttributeValue(page.getId(), att);
                 }
-            });
-            
+            }
+
+            for(long id : currentAttributes){
+                removeAttributeValue(page.getId(), id);
+            }
+
             conn.commit();
         } finally {
```

Comment:
We disable auto commit at the beginning of this method, since it is a multi transaction process to persist a single page in the database. There should be rollback if one of the transactions fails. After finishing persisting the page we want to use auto commit again, even if an error occurred. 

The other methods use auto commit since it is only a single transaction for every method.